### PR TITLE
[Java 21] Remove the use of SLF4J from Ballerina Runtime

### PR DESCRIPTION
**Pull request by ravinperera00**
## Approach
The suggested changes will refactor the code to remove the use of `SLF4J` library methods and classes from the Ballerina Runtime using other Java alternatives.

## Remarks
Planning to format the log output including the stacktrace to match the previous template later.

## Check List 
- [X] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/transactions/TransactionResourceManager.java [hunk 1]
```diff
@@ -618,4 +617,9 @@ public Object getTransactionRecord(BArray xid) {
             return null;
         }
     }
+
+    private void logError(String message, Exception err) {
+        outStream.println(LocalDateTime.now() + "ERROR\t" + message + "\n");
+        err.printStackTrace();
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/transactions/TransactionResourceManager.java:623]: Shall we replace with  out.println(Arrays.toString(e.getStackTrace()))

Comment (ravinperera00) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/transactions/TransactionResourceManager.java:623]: Got it. We will lose the formatting though, hope that is ok.

**Comment by warunalakshitha** [comment]
Can we remove from rt jar

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/transactions/TransactionResourceManager.java [hunk 2]
```diff
@@ -639,7 +639,7 @@ private String generateCombinedTransactionId(String transactionId, String transa
     public void notifyResourceFailure(String gTransactionId) {
         failedResourceParticipantSet.add(gTransactionId);
         // The resource excepted (uncaught).
-        LOG.info("Trx infected callable unit excepted id : " + gTransactionId);
+        outStream.println("Trx infected callable unit excepted id : " + gTransactionId);
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/transactions/TransactionResourceManager.java:642]: Lets make outSream capital letters

Comment (ravinperera00) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/transactions/TransactionResourceManager.java:642]: Got it.

**Pull request closed without merging**